pwiz_tools/Skyline/SkylineGraphs.cs

```diff
@@ -2159,44 +2159,57 @@ public SrmDocument ChangePeakBounds(SrmDocument document, IEnumerable<ChangedPea
                 document = document.BeginDeferSettingsChanges();
             }
 
-            var changedGroupIds = new HashSet<IdentityPath>();
-            var peptideChanges = new Dictionary<IdentityPath, ChangedPeakBoundsEventArgs>();
+            var changedGroupIds = new HashSet<Tuple<IdentityPath, string>>();
```

Comment:
I think this should be HashSet&lt;Tuple&lt;IdentityPath, MsDataFileUri>>
(and the Dictionary on the next line should also use MsDataFileUri instead of string)
You can trust the "Equals" and "GetHashCode" methods on MsDataFileUri to do the right thing-- no reason to convert to string.

Reply:
I changed it - originally I did consider that, but MsDataFileUri doesn't have Equals or GetHashCode. So it's not guaranteed that inherited classes will override the default implementation is it?